### PR TITLE
Ampstart 2.0

### PR DESCRIPTION
**Pull request by sebastianbenz**

### CODE REVIEW

File: lib/next-amp-demo/next-env.d.ts [hunk 1]
```diff
@@ -0,0 +1,2 @@
+/// <reference types="next" />
```
Comment (alankent) [inline lib/next-amp-demo/next-env.d.ts]: Should this file have a copyright message? (Your call.)

File: lib/next-amp/src/components/AmpAccess.tsx [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import * as React from 'react';
+import Head from 'next/head';
+import {AmpIncludeCustomElement} from '../../src-gen/';
+
+/**
+ * Renders an amp-access element.
```
Comment (alankent) [inline lib/next-amp/src/components/AmpAccess.tsx]: Should a comment be added here why the default generated source could not be used? Might be interesting to the reader and other people who have to maintain in the future.

Comment (alankent) [reply lib/next-amp/src/components/AmpAccess.tsx]: (And same for all other manually authored components.) That is, have documentation somewhere for maintainers as well as users? But an AMP expert might also be interested - they need to understand the relationship between the React markup and AMP markup as amp.dev documentation talks AMP markup, not React.

File: lib/next-amp/src/components/AmpForm.tsx [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+
+import * as React from 'react';
+import {FunctionComponent, FormHTMLAttributes} from 'react';
+import {AmpIncludeCustomElement} from '../../src-gen/';
+
+/**
+ * Renders an amp-form element.
+ *
+ * @param {
+  FormHTMLAttributes<HTMLFormElement>,
```
Comment (alankent) [inline lib/next-amp/src/components/AmpForm.tsx]: Should each line here be prefixed with "*" or must it be removed to be parsed correctly? (Minor)

File: prettier.config.d.ts [hunk 4]
```diff
@@ -0,0 +1,13 @@
+export declare const bracketSpacing: boolean;
```
Comment (alankent) [inline prettier.config.d.ts]: Copyright message needed? (I don't think so)

File: templates/travel/components/ActivitiesPanel.js [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+import DrinksSvg from './svg/DrinksSvg';
+import FashionSvg from './svg/FashionSvg';
+import FoodSvg from './svg/FoodSvg';
+import MusicSvg from './svg/MusicSvg';
+import NatureSvg from './svg/NatureSvg';
+import NightlifeSvg from './svg/NightlifeSvg';
+import ToursSvg from './svg/ToursSvg';
+import WaterSvg from './svg/WaterSvg';
+
+export default function ActivitiesPanel(props) {
```
Comment (alankent) [inline templates/travel/components/ActivitiesPanel.js]: Your other examples put the export default at the end. Do you want the AMP Start examples to follow that approach, or use this style of export on the function declaration? (This is for me to do on AMP Start demos, not for you to change here as this file will go away later as ampstart project will hold master copy.)

File: templates/travel/components/ActivityButton.js [hunk 6]
```diff
@@ -0,0 +1,20 @@
+export default function ActivityButton({name, title, children}) {
+  return (
+    <a
+      href='results?amp=1'
```
Comment (alankent) [inline templates/travel/components/ActivityButton.js]: Looks like all the double quotes I had have been replaced by single quotes. Is this on purpose? (Do you want all the AMP Start code converted over?) Is this a Prettier config file change thing?

File: lib/next-amp-demo/next-env.d.ts (hunk 1 above)
Comment (sebastianbenz) [reply lib/next-amp-demo/next-env.d.ts]: :-) I think we can skip this one  - it's auto generated as well

File: lib/next-amp-demo/pages/components/AmpIframe.tsx [hunk 7]
```diff
@@ -0,0 +1,42 @@
...
+    layout='responsive'
+    width='500'
+    height='281'
+    sandbox='allow-scripts allow-same-origin allow-popups'
+    allowfullscreen=''
+    frameborder='0'
+    src='https://player.vimeo.com/video/140261016'
+    className='my-iframe'
+  >
+    <amp-img
```
[the remark this answers is not in the document]
Comment (sebastianbenz) [reply lib/next-amp-demo/pages/components/AmpIframe.tsx]: done

File: lib/next-amp/README.md [hunk 8]
```diff
@@ -0,0 +1,322 @@
+# AMP Next.js Library
+
+**TLDR; **build AMP pages in Next.js without worrying about AMPHTML syntax.
```
[the remark this answers is not in the document]
Comment (sebastianbenz) [reply lib/next-amp/README.md]: changed

File: lib/next-amp/README.md [hunk 9]
```diff
@@ -0,0 +1,322 @@
...
+
+```
+<div data-amp-bind-text="hello"/>
+<button on="tap:AMP.setState({ "hello": "world" })">
+```
+
+State can be initialized via:
+
+```
+<AmpState id="myState">{{
```
[the remark this answers is not in the document]
Comment (sebastianbenz) [reply lib/next-amp/README.md]: not sure this matters for now, I'll rewrite the README once we get closer to a release.

File: lib/next-amp/src/components/AmpAccess.tsx (hunk 2 above)
Comment (sebastianbenz) [reply lib/next-amp/src/components/AmpAccess.tsx]: I'd plan to add this to the readme (or docs).

File: lib/next-amp/src/components/AmpForm.tsx (hunk 3 above)
Comment (sebastianbenz) [reply lib/next-amp/src/components/AmpForm.tsx]: Removed all JSDOC param declarations. This is redundant when using typescript

File: prettier.config.d.ts (hunk 4 above)
Comment (sebastianbenz) [reply prettier.config.d.ts]: this is auto generated - a glitch in the build system I couldn't figure out

File: templates/travel/components/ActivitiesPanel.js (hunk 5 above)
Comment (sebastianbenz) [reply templates/travel/components/ActivitiesPanel.js]: I think your approach is good. Not sure why I did it differently.

File: templates/travel/components/ActivityButton.js (hunk 6 above)
Comment (sebastianbenz) [reply templates/travel/components/ActivityButton.js]: Prettify will auto fix these with every commit.

**Comment by sebastianbenz** [comment]
I've addressed your comments. PTAL

File: babel.config.js [hunk 10]
```diff
@@ -1,5 +1,5 @@
 /**
- * Copyright 2017 The AMP Start Authors. All Rights Reserved.
+ * Copyright 2019 The AMP HTML Authors. All Rights Reserved.
```
Comment (alankent) [inline babel.config.js]: Legal approved of this change I assume? Sometimes changing copyright has legal ramifications (but I doubt it matters here).

File: components/lightbox/template-preview.snip.html [hunk 11]
```diff
@@ -1,97 +0,0 @@
-{{!
```
Comment (alankent) [inline components/lightbox/template-preview.snip.html]: I assume you delete the "components" directory on purpose (they are not related to the templates)

File: lib/next-amp/README.md [hunk 12]
```diff
@@ -0,0 +1,322 @@
+# AMP Next.js Library
```
Comment (alankent) [inline lib/next-amp/README.md]: I am not checking all this - I assume identical to the separate repo you used to have.

File: templates/travel/components/TravelHero.js [hunk 13]
```diff
@@ -0,0 +1,237 @@
+import TravelHeroSearch from '../components/TravelHeroSearch';
```
Comment (alankent) [inline templates/travel/components/TravelHero.js]: Once accepted, should I merge all my changes into this master repo and refork to keep working? Or should we delete everything from here (the Travel template) and reimport it?

File: components/lightbox/template-preview.snip.html (hunk 11 above)
Comment (sebastianbenz) [reply components/lightbox/template-preview.snip.html]: yes, we can re-add it later

File: babel.config.js (hunk 10 above)
Comment (sebastianbenz) [reply babel.config.js]: I'm using AMP HTML authors for everything AMP related.

File: templates/travel/components/TravelHero.js (hunk 13 above)
Comment (sebastianbenz) [reply templates/travel/components/TravelHero.js]: I'd suggest you create PR against the amstart-2.0 branch

File: lib/next-amp/README.md (hunk 12 above)
Comment (sebastianbenz) [reply lib/next-amp/README.md]: yup

**Pull request merged**